cmd/subnetcmd/publish.go

```diff
@@ -0,0 +1,380 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package subnetcmd
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/spf13/cobra"
+
+	"github.com/ava-labs/apm/types"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/subnet"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	alias          string
+	repoURL        string
+	vmDescPath     string
+	subnetDescPath string
+)
+
+type newPublisherFunc func(string, string, string) subnet.Publisher
+
+// avalanche subnet deploy
+func newPublishCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "publish [subnetName]",
+		Short:        "Publish the subnet's VM to a repository",
+		Long:         ``,
+		SilenceUsage: true,
+		RunE:         publish,
+		Args:         cobra.ExactArgs(1),
+	}
+	cmd.Flags().StringVar(&alias, "alias", "", "An alias for the repo to publish to")
+	cmd.Flags().StringVar(&repoURL, "repo-url", "", "The URL of the repo where we are publishing")
+	cmd.Flags().StringVar(&vmDescPath, "vm-file-path", "", "Path to the VM description file. If not given, a prompting sequence will be initiated.")
+	cmd.Flags().StringVar(&subnetDescPath, "subnet-file-path", "", "Path to the Subnet description file. If not given, a prompting sequence will be initiated.")
```

Comment:
`local-only`? `no-push`? 

`offline-only` doesn't tell me much, personally. Using `local-only` for now, feel free to suggest others.

Reply:
Note that this creates problems with the new  `isAlreadyPublished` check, which would return published because it would find the subnetName in the local repo structure.

TBH I am not so sure this is a good idea to have this.

Reply:
Instead of creating the local directory in the internal structure, should we just write it to a directory of the user's choice or `/tmp`?

Reply:
The process I'm thinking of here is that the user has a weird git setup and doesn't want to manage pushes through our tool. They just want the files to be generated with the right format and they'll setup the git repo themselves.

Reply:
I feel this might create more confusion. Also, nobody actually needs to know this is using git underneath. Everybody _thinks_ to know git, if we hand out control we indeed loose control of what's going on altogether and can't manage any state properly I am afraid.

For the process you are thinking of, it might be best to still do `local-only` (no push), then have the user grab the files and put/push them wherever/however they want.

Reply:
Yes, I think that's what I'm suggesting